### PR TITLE
[Refactor] Using MMCls backbones

### PR DESCRIPTION
**Pull request by fangyixiao18**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Using backbones from mmcls, 

## Modification

modify resnet.py
modify resnext.py

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by Jiahao000 — APPROVED** [review]

File: mmselfsup/models/backbones/vision_transformer.py [hunk 1]
```diff
@@ -5,7 +5,8 @@
 
 import torch.nn as nn
 from mmcls.models.backbones import VisionTransformer as _VisionTransformer
-from mmcls.models.utils import PatchEmbed, to_2tuple
```
Comment (YuanLiuuuuuu) [inline mmselfsup/models/backbones/vision_transformer.py:8]: I think this modification should be in a new PR.

File: mmselfsup/models/backbones/resnext.py [hunk 2]
```diff
@@ -1,165 +1,56 @@
...
-                width_per_group=width_per_group,
-                style=style,
-                with_cp=with_cp,
-                conv_cfg=conv_cfg,
-                norm_cfg=norm_cfg))
-
-    return nn.Sequential(*layers)
-
-
 @BACKBONES.register_module()
```
Comment (ZwwWayne) [inline mmselfsup/models/backbones/resnext.py:9]: Why do we still need ResNeXt? Can we also import `mmcls.ResNeXt`?

Comment (fangyixiao18) [reply mmselfsup/models/backbones/resnext.py:9]: > Why do we still need ResNeXt? Can we also import `mmcls.ResNeXt`?

As we are using forward function from our own ResNet, which is different from forward function in mmcls.

Comment (fangyixiao18) [reply mmselfsup/models/backbones/resnext.py:9]: > Why do we still need ResNeXt? Can we also import `mmcls.ResNeXt`?

As we are using forward function from our own ResNet, which is different from forward function in mmcls.

Comment (YuanLiuuuuuu) [reply mmselfsup/models/backbones/resnext.py:9]: > Why do we still need ResNeXt? Can we also import `mmcls.ResNeXt`?

MMCls does not output the feature map from the first `conv` layer, which we will use for downstream evaluation. That's why we rewrite the `forward` function.

**Review by YuanLiuuuuuu — APPROVED** [review]

File: mmselfsup/models/backbones/resnext.py (hunk 2 above)
Comment (ZwwWayne) [reply mmselfsup/models/backbones/resnext.py:9]: 1. We should add comments in the code to explain the difference and the reason. 
2. We can give feedback to mmcls, e.g., create a feature request. Invite @mzr1996 to take a look.

**Pull request merged**